### PR TITLE
Mirror manifests

### PR DESCRIPTION
**Pull request by dmage**

### CODE REVIEW

**Comment by dmage** [comment]
@legionus ptal

File: pkg/dockerregistry/server/imagestream.go [hunk 1]
```diff
@@ -170,3 +171,39 @@ func (is *imageStream) rememberLayersOfImage(ctx context.Context, image *imageap
...
+	if err != nil {
+		return false, err
+	}
+
+	if len(tag) == 0 {
+		// if the client pulled by digest, find the corresponding tag in the image stream
+		tag, _ = imageapiv1.LatestImageTagEvent(stream, dgst.String())
+	}
+
+	if insecure, _ := stream.Annotations[imageapi.InsecureRepositoryAnnotation]; insecure == "true" {
```
Comment (legionus) [inline pkg/dockerregistry/server/imagestream.go]: Looks like this condition uses only `stream`. Please move it before `if len(tag) == 0`.

File: pkg/dockerregistry/server/util.go [hunk 2]
```diff
@@ -69,17 +68,12 @@ func wrapKStatusErrorOnGetImage(repoName string, dgst digest.Digest, err error)
 
 // getImportContext loads secrets for given repository and returns a context for getting distribution clients
 // to remote repositories.
-func getImportContext(
```
Comment (miminar) [inline pkg/dockerregistry/server/util.go]: *nit*: doc: the repository is no longer given

File: pkg/dockerregistry/server/remoteblobgetter.go [hunk 3]
```diff
@@ -29,13 +30,37 @@ type BlobGetterService interface {
 type imageStreamGetter func() (*imageapiv1.ImageStream, error)
 type secretsGetter func() ([]kapiv1.Secret, error)
 
+type digestBlobStoreCache struct {
```
Comment (miminar) [inline pkg/dockerregistry/server/remoteblobgetter.go]: Please make a note to make this cache being used globally.

**Comment by bparees** [comment]
no comments from me, looks reasonable given my level of understanding.

**Comment by dmage** [comment]
@legionus @miminar comments have been addressed.

**Pull request merged**